libraries/freertos_plus/standard/pkcs11/include/iot_pkcs11.h

```diff
@@ -0,0 +1,233 @@
+ * headers.
+ */
+
+#define CK_PTR          *
+
+#ifndef NULL_PTR
+    #define NULL_PTR    0
+#endif
+
+#define CK_DEFINE_FUNCTION( returnType, name )             returnType name
+#define CK_DECLARE_FUNCTION( returnType, name )            returnType name
+#define CK_DECLARE_FUNCTION_POINTER( returnType, name )    returnType( CK_PTR name )
+#define CK_CALLBACK_FUNCTION( returnType, name )           returnType( CK_PTR name )
+
+/**
+ *   @brief Length of a SHA256 digest, in bytes.
+ */
+#define pkcs11SHA256_DIGEST_LENGTH           32
+
+/**
+ * @brief Length of a curve P-256 ECDSA signature, in bytes.
+ */
+#define pkcs11ECDSA_P256_SIGNATURE_LENGTH    64
+
+ /**
+ * @brief Public exponent for RSA - 2048
+ */
+#define pkcs11RSA_2048_PUBLIC_EXPONENT       { 0x01, 0x00, 0x01 }
+
+/**
+ * @brief Length of PKCS #11 signature for RSA 2048 key, in bytes.
+ */
+#define pkcs11RSA_2048_SIGNATURE_LENGTH      ( 2048 / 8 )
+
+/**
+ * @brief Length of RSA signature data before padding.
+ *
+ * This is calculated by adding the SHA-256 hash len (32) to the 19 bytes in
+ * pkcs11STUFF_APPENDED_TO_RSA_SIG = 51 bytes total.
+ */
+#define pkcs11RSA_SIGNATURE_INPUT_LENGTH     51
+
+/**
+ * @brief Elliptic-curve object identifiers.
+ * From https://tools.ietf.org/html/rfc6637#section-11.
+ */
+#define pkcs11ELLIPTIC_CURVE_NISTP256        "1.2.840.10045.3.1.7"
+
+/**
+ * @brief Maximum length of storage for PKCS #11 label, in bytes.
+ */
+#define pkcs11MAX_LABEL_LENGTH               32 /* 31 characters + 1 null terminator. */
+
+/**
+ * @brief OID for curve P-256.
+ */
+#define pkcs11DER_ENCODED_OID_P256           { 0x06, 0x08, 0x2a, 0x86, 0x48, 0xce, 0x3d, 0x03, 0x01, 0x07 }
+
+
+/**
+ * @brief Set to 1 if importing private keys is supported.
+ *
+ * If private key import is not supported, this value should be defined 0 in aws_pkcs11_config.h
+ */
+#ifndef pkcs11configIMPORT_PRIVATE_KEYS_SUPPORTED
+    #define pkcs11configIMPORT_PRIVATE_KEYS_SUPPORTED    1
+#endif
+
+/*
+ * DigestInfo :: = SEQUENCE{
+ *      digestAlgorithm DigestAlgorithmIdentifier,
+ *      digest Digest }
+ *
+ * DigestAlgorithmIdentifier :: = AlgorithmIdentifier
+ * Digest :: = OCTET STRING
+ *
+ * This is the DigestInfo sequence, digest algorithm, and the octet string/length
+ * for the digest, without the actual digest itself.
+ */
+#define pkcs11STUFF_APPENDED_TO_RSA_SIG    { 0x30, 0x31, 0x30, 0x0d, 0x06, 0x09, 0x60, 0x86, 0x48, 0x01, 0x65, 0x03, 0x04, 0x02, 0x01, 0x05, 0x00, 0x04, 0x20 }
+
+/* Bring in the public header. */
+
+/* Undefine the macro for Keil Compiler to avoid conflict */
+#if defined( __PASTE ) && defined( __CC_ARM )
+    /* ARM RCVT stdint.h has a duplicate definition with PKCS #11. */
+    #undef __PASTE
+#endif
+
+#include "pkcs11.h"
+
+/* Elliptic Curve Private Key Template
+ * The object class must be the first attribute in the array.
+ * https://www.cryptsoft.com/pkcs11doc/v220/group__SEC__12__3__4__ELLIPTIC__CURVE__PRIVATE__KEY__OBJECTS.html */
```

Comment:
Could we reference the standard directly instead? This is obviously the website of a for-profit company.